### PR TITLE
Ensure that macOS downstream builds do not hardcode a specific minor.patch version when linking to the library

### PR DESCRIPTION
**Pull request by traversaro**
Fix an issue in SDL2_image  similar to https://github.com/conda-forge/sdl2_ttf-feedstock/issues/18 by backporting https://github.com/libsdl-org/SDL_ttf/pull/349 and adding symlinks for existing builds.

There is currently no issue upstream for SDL_image, I will track this upstream following the outcome of https://github.com/libsdl-org/SDL_ttf/pull/349 .

This also fixes the compilation of sdl2_image with a recent CMake, backporting https://github.com/libsdl-org/SDL_image/pull/439 . This should also unblock the update to recent versions of sdl2_image (see https://github.com/conda-forge/sdl2_image-feedstock/pull/31).

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
Cool, environment regression on Windows, either from libtiff or CMake I guess:
~~~
-- Found TIFF: D:/bld/sdl2_image_1711371034934/_h_env/Library/lib/cmake/tiff (found version "4.6.0")
CMake Error at cmake/PrivateSdlFunctions.cmake:247 (message):
  TIFF::TIFF is not a shared library, but has type=INTERFACE_LIBRARY
-- Dynamic libtiff: $<TARGET_FILE_NAME:TIFF::TIFF>
Call Stack (most recent call first):
-- SDL2_image: Using system libwebp
~~~

**Comment by traversaro** [comment]
> Cool, environment regression on Windows, either from libtiff or CMake I guess:
> 
> ```
> -- Found TIFF: D:/bld/sdl2_image_1711371034934/_h_env/Library/lib/cmake/tiff (found version "4.6.0")
> CMake Error at cmake/PrivateSdlFunctions.cmake:247 (message):
>   TIFF::TIFF is not a shared library, but has type=INTERFACE_LIBRARY
> -- Dynamic libtiff: $<TARGET_FILE_NAME:TIFF::TIFF>
> Call Stack (most recent call first):
> -- SDL2_image: Using system libwebp
> ```

I guess this is https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8803 .

**Comment by traversaro** [comment]
Cross-compiled builds are slow, but they have already succeeded and the following changes are unrelated. So I think the PR is ready for review @conda-forge/sdl2_image , thanks!

**Comment by traversaro** [comment]
> > Cool, environment regression on Windows, either from libtiff or CMake I guess:
> > ```
> > -- Found TIFF: D:/bld/sdl2_image_1711371034934/_h_env/Library/lib/cmake/tiff (found version "4.6.0")
> > CMake Error at cmake/PrivateSdlFunctions.cmake:247 (message):
> >   TIFF::TIFF is not a shared library, but has type=INTERFACE_LIBRARY
> > -- Dynamic libtiff: $<TARGET_FILE_NAME:TIFF::TIFF>
> > Call Stack (most recent call first):
> > -- SDL2_image: Using system libwebp
> > ```
> 
> I guess this is https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8803 .

Fixed by backporting https://github.com/libsdl-org/SDL_image/pull/439 .

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,6 +10,8 @@ source:
   patches:
     - patches/0001-look-for-conda-forge-variant-of-libwebp.patch                # [win]
     - patches/0002-BLD-Use-CMake-artifacts-instead-of-custom-findwebp.patch     # [win]
+    - patches/0003-fix-library-soversion-name-in-macos.patch                    # [osx]
+    - patches/0004-fix-libtiff-linking-with-cmake328.patch
```
Comment (h-vetinari) [inline recipe/meta.yaml]: Please document any relevant references for patches you add. Something like:
```suggestion
    # equivalent to https://github.com/libsdl-org/SDL_ttf/pull/349 for fixing SOVER
    - patches/0003-fix-library-soversion-name-in-macos.patch                    # [osx]
    # backport https://github.com/libsdl-org/SDL_image/pull/439
    - patches/0004-fix-libtiff-linking-with-cmake328.patch
```

**Review by h-vetinari — APPROVED** [review]
Thanks a lot for fixing this, and for chasing down the tiff issue! 🙏 

I only have one nit, then this is ready to go!

File: recipe/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipe/meta.yaml]: Thanks, suggestion added!

**Pull request merged**